src/firmware_new/boards/frankie_v1/io/infineon_TLE5009_E1000.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include <math.h>
+/**
+ * This file is an abstraction around the Infineon TLE5009 E1000 GMR angle sensor
+ *
+ * Datasheet:
+ * https://www.infineon.com/dgdl/Infineon-TLE5009_FDS-DataSheet-v01_01-en.pdf?fileId=db3a304330f686060131421d8ddd56b0
+ */
+
+typedef struct InfineonTLE5009E1000 InfineonTLE5009E1000_t;
+
+/**
+ * Creates an infineon TLE5009 E1000 abstraction given the calibration parameters
+ *
+ * @param x_max_value [in] The maximum value observed over a full period of the X output
+ * waveform
+ *
+ * @param x_min_value [in] The minimum value observed over a full period of the X output
+ * waveform
+ *
+ * @param y_max_value [in] The maximum value observed over a full period of the Y output
+ * waveform
+ *
+ * @param y_min_value [in] The minimum value observed over a full period of the Y output
+ * waveform
+ *
+ * @param x_magnitude_45_degrees [in] The X waveform magnitude at 45 degrees (as read by
```

Comment:
super nit: We mention 45 and 135 degrees here without mention of a physical reference (clockwise, counterclockwise), or a reference to zero angle. I'm assuming it doesn't really matter since the zero position is arbitrary and everything is relative to that, but I'm also not sure if the calibration parameters need to be following the same convention and the output / interface of this abstraction.

(ie. if we calibrate with 45 degrees being clockwise rotation, and the abstraction treats 45 degrees as counterclockwise, can things be less accurate / messed up?)

Reply:
The reference is 'as read by the angle sensor'. There is a was to get away with not using an external reference described in the data sheet.

The calibration values there are intended to be the 'mean' values which are equal to the weighted sum of the CW and CCW calibration values.

I've added a comment to make this more clear